submissions/kasionio/oop-exercise/index.js

```diff
@@ -0,0 +1,100 @@
+/* Refer to https://github.com/OleksiyRudenko/a-tiny-JS-world for the task details
+   Complete the below for code reviewers' convenience:
+
+   Code repository: https://github.com/kasionio/a-tiny-JS-world/tree/gh-pages
+   Web app: https://kasionio.github.io/a-tiny-JS-world
+   */
+
+// ======== OBJECTS DEFINITIONS ========
+// Define your objects here
+class Inhabitant {
+  constructor(species, name, gender, legs, saying, friends = 'forever alone') {
+      this.species = species;
+      this.name = name;
+      this.gender = gender;
+      this.legs = legs;
+      this.saying = saying;
+      this.friends = friends;
+  }
+
+  addFriends(...friends)  {
+    this.friends = [...friends].map(friend => friend.name).join(', ');
```

Comment:
If I do e.g. `cat.addFriends(dog, man); cat.addFriends(woman, dog);` will friends be really **added**?

What will happen if I do e.g. `cat.addFriends([dog, man])` instead of `cat.addFriends(dog, man)`? Will the code work as expected?

If we pack method arguments in an array (`addFriends(...friends) {...}`) why do we need unpacking array and packing it back into array (`[...friends]`) in the function body?

Reply:
Thanks! Now I realize that my code just made the output but wasn't really adding friends. Fixed the logic. Test in the browser console. works fine now :)